main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -1645,7 +1645,7 @@ private Waypoint findWaypoint(final Waypoint searchWp) {
             if (!StringUtils.isBlank(searchWpName)) {
                 final String wpName = waypoint.getName();
                 final WaypointType wpType = waypoint.getWaypointType();
-                if (searchWpName.equals(wpName) && searchWp.getWaypointType().getL10n().equals(wpType.getL10n())) {
+                if (searchWpName.equals(wpName) && (searchWp.getWaypointType().getNameForNewWaypoint().equals(wpType.getNameForNewWaypoint()) || searchWp.getWaypointType().getL10n().equals(wpType.getL10n()))) {
```

Comment:
IMHO that should not be a problem here, as a full string comparison ("equals") is done here, whereas above it's a "contains". (But, actually, no need to compare both the old name and the new name, as it's about waypoint types, not names. => Will update that in my PR.)

But I wonder why we do not compare simply the type id instead of type names with each other? something like `searchWp.getWp.getWaypointType().id.equals(wpType.id)` - any reason not to?

Reply:
@moving-bits you are right, didn't see that. I also don't understand why names are compares instead of ids. I think this would be better actually. @murggel any reason for this?

Reply:
In the PR I have reverted to the original code for this method for now. If no one objects, I can switch comparison to id, and do some other optimization (eg: why do a `searchWp.getName()` within the loop etc.) - not that important, as it's a small loop, just stumbled upon while checking for waypoint type name requests.

Reply:
To be honest, I don't know anymore, why I used the name there and not the id... probably to avoid duplicate names of waypoint, if two types have different ids, but same name?

Reply:
Ok, thanks for replying. This is the waypoint _type_ id, which should not change, not the waypoint id. I'll create a PR.

Reply:
Hm, I think there was a problem with waypoint-type OWN an WAYPOINT. To avoid a duplicate here, if a waypoint with same name exists, I checked the names. But probably this check should be done differently.. Different waypoint-type-id would be not sufficient ...